### ISSUE TITLE
[Reject] [Tracking] Reject (fka Decline) Expense in NewDot

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#52049]
[DESIGN DOC](https://docs.google.com/document/d/1GPraB4x5B2DTKnS2zMhjYLOzhPp3fjXp-WAAxEXYOt8/edit?tab=t.0)
[Customer Notification Issue](https://github.com/Expensify/App/issues/56442)

**Note:** we switched from Decline to Reject midstream. The old doc referencing Decline is [here](https://docs.google.com/document/d/1ukHoMDoX07lwstLn7fBKjWnnopLkDbezQv6Yd6hLIuo/edit?tab=t.0#heading=h.71gukbqqon31).

## Proposal
**[WN Post](https://expensify.slack.com/archives/CC7NECV4L/p1730997293017289)**

**Proposal:** Allow approvers to reject individual expenses in NewDot

**Background:** Generally, users have full control over the expenses assigned to them for processing. Submitters can create, edit, submit, or delete expenses to manage their reports effectively. Approvers typically have the ability to act on expenses by approving, editing, or rejecting them, aligning with the general expectation that individuals can manage tasks assigned to them. This level of control ensures efficiency and meets user expectations based on their use of Expensify Classic and expense reporting processes across the industry.

**Problem:** When submitted expenses just don’t fit within expense policy guidelines and will never be approved or paid, approvers are unable to remove them from their inbox. This prevents them from effectively and efficiently managing the tasks in their inbox, causing frustration and confusion.

**Solution:** Add the ability for approvers to unreport individual expenses submitted to them in NewDot. The naming of this function is TBD, but I’ll use ‘reject’ for familiarity’s sake. At a high level, let’s do the following:
- Add a ‘Reject’ button to the expense details page
- When an approver taps the button, we’ll show a confirmation page where we explain the difference between reject and hold, and allow the approver to confirm rejection with a comment or cancel.
- When the approver confirms rejection with a comment, we’ll remove the expense from the report, and the submitter will be able to access it in their Self DM.

This solution aligns NewDot with functionality approvers are accustomed to in OldDot and gives approvers full control of the expenses in their inbox. 

## Implementation
**Frontend:**
- https://github.com/Expensify/App/issues/58697
- https://github.com/Expensify/App/pull/62787
- https://github.com/Expensify/App/pull/65239

**Backend:**
- https://github.com/Expensify/Auth/pull/15941
- https://github.com/Expensify/Auth/pull/16611
- https://github.com/Expensify/Expensify/issues/523001
- https://github.com/Expensify/Expensify/issues/482350
- https://github.com/Expensify/Expensify/issues/482349
- https://github.com/Expensify/Expensify/issues/482351
- https://github.com/Expensify/Web-Expensify/pull/48100

**Follow up issues:**
- https://github.com/Expensify/App/issues/70482
- https://github.com/Expensify/App/issues/70483
- https://github.com/Expensify/App/issues/70484
- https://github.com/Expensify/App/issues/63363

## Tasks

- [x] Post **Proposal (full Problem/Solution statement)** in `#expensify-open-source`
- [x] Wait at least one full business day, and until the post has a majority (2/3) of positive reactions (👍)
- [x] Paste Proposal in the space above with a link to the Slack thread
- [x] Email `strategy@expensify.com` and paste in the Proposal
- [x] Fill out the **High-level overview of the problem**, **Timeline**, and **Terminology** sections of the Design Doc
- [x] Email `strategy@expensify.com` (continue the same email chain as before) with the link to your Design Doc
- [x] Host a **pre-design** meeting ([example](https://expensify.slack.com/archives/C01GTK53T8Q/p1665794669539419)) in `#expensify-open-source` to discuss any necessary details in public before filling out the **High-level of proposed solution** section.
- [x] Fill out the **High-level of proposed solution** section
- [x] Email `stategy@expensify.com` again with links to the doc and pre-design conversation in Slack
- [x] Add the `DesignDocReview` label to get the **High-level of proposed solution** section reviewed
- [x] Respond to any questions or concerns and bring up blockers in Slack to get a consensus if necessary
- [x] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [x] Host another pre-design meeting in `#expensify-open-source` to ask for engineering feedback on the technical solution.
- [x] Fill out the **Detailed implementation of the solution** and related sections.
- [x] Re-add the `DesignDocReview` label to this issue
- [ ] Respond to any questions or concerns and bring up blockers in Slack to get consensus if necessary
- [ ] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [ ] Email `strategy@expensify.com` one last time to let them know the Design Doc is moving into the implementation phase
- [ ] Implement the changes
- [ ] Add regression tests so that QA can test your feature with every deploy ([instructions](https://stackoverflowteams.com/c/expensify/questions/363))
- [ ] Send out a follow up email to `strategy@expensify.com` once everything has been implemented and do a **Project Wrap-Up** retrospective that provides:
    - Summary of what we accomplished with this project
    - What went well?
    - What could we have done better?
    - What did we learn?

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by garrettmknight** [comment]
Started to mock out the pre-design.

**Comment by garrettmknight** [comment]
Same as above

**Comment by garrettmknight** [comment]
Worked through some [mocks](https://www.figma.com/design/ucnG7phAKyox7EMTDAoMhD/Decline-Expense?node-id=1-85&node-type=canvas&t=fuNqhPmqHsrzau58-0) and the DD

**Comment by garrettmknight** [comment]
I think I could be ready for the pre-design early next week.

**Comment by garrettmknight** [comment]
Still making progress pre-design by Weds.

**Comment by garrettmknight** [comment]
Aiming to have a very small predesign tomorrow afternoon.

**Comment by garrettmknight** [comment]
Made a bunch of progress today - still need to propose a few more details using the new Hold RHP.

**Comment by garrettmknight** [comment]
[Pre-design here, finally.](https://expensify.slack.com/archives/C06ML6X0W9L/p1733851713458989)

**Comment by garrettmknight** [comment]
We're folding this into the [action button project](https://github.com/Expensify/Expensify/issues/447024) so I'm going to close this GH and track there instead.

**Comment by garrettmknight** [comment]
Unfolding from the action button project. This is still tied and we should complete it as part of the action button functions. I'll get the HL DD out early next week!

**Comment by garrettmknight** [comment]
Everything but the data storage section is complete - gonna try to recruit an engineer tomorrow.

**Comment by garrettmknight** [comment]
Same as above!

**Comment by garrettmknight** [comment]
Finishing up Figmas and working with @Beamanator on the data storage section.

**Comment by garrettmknight** [comment]
Still making progress, chatted with Beaman. Found an interesting backward compatibility issue today: https://expensify.slack.com/archives/C06ML6X0W9L/p1736787149906839

**Comment by garrettmknight** [comment]
Almost ready, just missing some small adds to the data storage section.

**Comment by garrettmknight** [comment]
Sounds like @Beamanator is gonna finish off the data storage section tomo and then we'll be good to go.

**Comment by garrettmknight** [comment]
Alright, we're good to goooooooo.

**Comment by garrettmknight** [comment]
Dropping to weekly while we get our reveiws.

**Comment by jliexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by shawnborton** [comment]
I have read and reviewed this Design Doc!

**Comment by cjjohnson121** [comment]
I have read and reviewed this Design Doc!

**Comment by MonilBhavsar** [comment]
I have read and reviewed this Design Doc!

**Comment by lydiabarclay** [comment]
I have read and reviewed this Design Doc!

**Comment by dylanexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by maddylewis** [comment]
I have read and reviewed this Design Doc!

**Comment by trjExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by MitchExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by laurenreidexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by garrettmknight** [comment]
Added Invoicing to Out of Scope.

**Comment by JmillsExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by davidcardoza** [comment]
I have read and reviewed this Design Doc!

**Comment by garrettmknight** [comment]
A few discussions about copy, but we're ready for a detailed author. Recruiting!

**Comment by laurenreidexpensify** [comment]
@garrettmknight was this meant to be listed under retain?

**Comment by garrettmknight** [comment]
Made a couple small updates to include updating OldDot to use Decline. Now we're just ready for the detailed when @lakchote has some time.

**Comment by lakchote** [comment]
Had the time today to read the Design Doc in details, especially the Data Storage part which will be crucial for the technical implementation. I should be able to work on the first bits of the detailed doc tomorrow.

**Comment by DylanDylann** [comment]
@lakchote @garrettmknight  It seems we already require expert contributors to implement, if this would necessitate a C+ for review please let me know. I am enthusiastic about participating in this project. I already worked a lot with the expense flow and also have experience when working on other projects

**Comment by ryanschaffer** [comment]
I have read and reviewed this Design Doc!

**Comment by lakchote** [comment]
Couldn't make much progress today as my environment is not working properly since latest VM changes on Friday ([Slack](https://expensify.slack.com/archives/C07J32337/p1740402136060669?thread_ts=1732198641.569679&cid=C07J32337))

**Comment by lakchote** [comment]
I've been making progress on the frontend implementation section today. 

I've also wrote it in a way to make it as easy to read as possible by avoiding putting too much code in there.

**Comment by DylanDylann** [comment]
@garrettmknight Could you help to check this [comment](https://github.com/Expensify/App/issues/52049#issuecomment-2674985107)? 

> I've been making progress on the frontend implementation section today.

Could I help you with anything from this time?

**Comment by lakchote** [comment]
> [@garrettmknight](https://github.com/garrettmknight) Could you help to check this [comment](https://github.com/Expensify/App/issues/52049#issuecomment-2674985107)?
> 

@DylanDylann @garrettmknight was OOO that's why he didn't answer you, he should be back today!

> > I've been making progress on the frontend implementation section today.
> 
> Could I help you with anything from this time?

Thanks for proposing your help. I should be able to complete soon the frontend section, and I'd love to have your opinion on it when it's ready, as you have worked a bit on the expense flow. The detailed implementation is intentionnally not filled with much code, as a good design doc won't have any code and would just contain descriptions of what needs to be done in the logic. 

I'll tag you when it's available @DylanDylann if you're up for an initial re-review! That way we can adjust if needed.

**Comment by lakchote** [comment]
Frontend section is done now, Offline Support section is done too. 
 
I'll be switching to the backend side of things.

**Comment by robertjchen** [comment]
I have read and reviewed this Design Doc!

**Comment by lydiabarclay** [comment]
I have read and reviewed this Design Doc!

**Comment by lakchote** [comment]
@lydiabarclay you've added your name to the HL review of the Design doc and not the Detailed section

![Image](https://github.com/user-attachments/assets/1c3b50cf-e056-4dd3-8e41-98a1174930d0)

Could you please make the adjustments? Thank you!

**Comment by lydiabarclay** [comment]
@lakchote done!

**Comment by aldo-expensify** [comment]
I have read and reviewed this Design Doc!

**Comment by davidcardoza** [comment]
I have read and reviewed this Design Doc!

**Comment by MitchExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by garrettmknight** [comment]
Looks like we've got our reviews! @lakchote you ready to get the implementation issues going? Anything I can do to help?

**Comment by lakchote** [comment]
> Looks like we've got our reviews! [@lakchote](https://github.com/lakchote) you ready to get the implementation issues going? Anything I can do to help?
>
@garrettmknight since I'll be OOO next week, I need to find someone volunteering for taking care of things while I'm away. 
I've done some outreach and will report back. If I can't find someone, leveraging your help will be greatly appreciated!

I have created all the issues already.

Frontend:
- [Implement the Decline Money Request flow](https://github.com/Expensify/App/issues/58697#top)


Backend:
- [[Decline Money in NewDot] Implement the MarkTransactionViolationAsResolved command](https://github.com/Expensify/Expensify/issues/482350#top)
- [[Decline Money in NewDot] Implement the DeclineMoneyRequest command](https://github.com/Expensify/Expensify/issues/482349#top)
- [[Decline Money in NewDot] Add the NVP dismissedDeclineUseExplanation](https://github.com/Expensify/Expensify/issues/482351#top)

**Comment by MariaHCD** [comment]
I have read and reviewed this Design Doc!

**Comment by garrettmknight** [comment]
@youssef-lr is working on the first BE issue!

**Comment by youssef-lr** [comment]
@garrettmknight can we add this project to the CAP software?

**Comment by garrettmknight** [comment]
This project didn't qualify since it's pretty much a straight port over from Classic.

**Comment by lakchote** [comment]
Not overdue. Progress is still happening.

**Comment by garrettmknight** [comment]
Still making progress!

**Comment by garrettmknight** [comment]
We've got all the BE merged, FE comes next.

**Comment by lakchote** [comment]
Not overdue, progress is still happening.

**Comment by mananjadhav** [comment]
@lakchote A few things:

1. I can see there's an existing UI for More menu but it's behind the beta - `canUseTableReportView`. I am reusing the same beta to avoid the rework on the UI.

2. I need the assets - Thumbs down for button, illustration for the modal and also for the section menu.

3. I've added `CONST.REPORT.SECONDARY_ACTIONS.DECLINE` to [getSecondaryReportActions](https://github.com/Expensify/App/blob/f3a347010e8bb0eaf313d12947fc73828e0f7783/src/libs/ReportSecondaryActionUtils.ts#L454-L456). But do we need additional checks like `isDeleteAction`.

4. If the user dismisses the hold educational modal by clicking it outside the modal, I remember we still discussed we'll set the `nvp_dismissedHoldUseExplanation` as true. But do we still open the decline request RHP?

https://github.com/user-attachments/assets/9a83ca7a-1a78-41b8-853c-14b0793533a2

**Comment by mananjadhav** [comment]
quick bump @garrettmknight and @dannymcclain

**Comment by garrettmknight** [comment]
> Add a ‘Decline` option as a secondary action on individual expenses for approvers/payers only.
@mananjadhav the option should only be available to approvers if they're assigned (not admins when a report is Processing) and payers once a report is approved.

**Comment by dannymcclain** [comment]
@mananjadhav assets for ya!

- **Stopwatch illustration:** `App/assets/images/simple-illustrations/simple-illustration__stopwatch.svg`
- **X illustration**: [simple-illustration__generic-no.svg.zip](https://github.com/user-attachments/files/20105974/simple-illustration__generic-no.svg.zip)
- **Modal illustration:** [CardIllustration-Hold-vs-Decline.svg.zip](https://github.com/user-attachments/files/20105981/CardIllustration-Hold-vs-Decline.svg.zip)
- **Thumbs down icon:** [Thumbs Down.svg.zip](https://github.com/user-attachments/files/20105985/Thumbs.Down.svg.zip)

**Comment by mananjadhav** [comment]
@dannymcclain I think we agreed not to use  ❌ right? Can you confirm the icon please? or share the appropriate thumbs down illustration?

<img width="925" alt="Image" src="https://github.com/user-attachments/assets/2893c158-1e6f-435d-94c9-054b685c2a12" />

Also when I compare the empty state hold expense and the modal illustration, I see a difference in the image width. Is that expected?

https://github.com/user-attachments/assets/bfeccc6e-60fc-4029-a003-182e1d4993d7

Should we update the illustration or fix it with some styling in the code?

<img width="703" alt="Image" src="https://github.com/user-attachments/assets/3d9ae344-858c-4b7f-91d8-9256abeb4536" />

**Comment by mananjadhav** [comment]
Thanks these new illustrations work fine. I am unblocked now and reverting the extra changes.

**Comment by mananjadhav** [comment]
I finished with all the UI. Now adding the checks for the action and the API intgeration.

**Comment by garrettmknight** [comment]
@mananjadhav how we tracking here? When do you expect we'll be able to start testing?

**Comment by garrettmknight** [comment]
Sounds good, thanks for the hustle. Let me know if there's anything we can do to unblock or answer questions along the way.

**Comment by garrettmknight** [comment]
@mananjadhav trying to set some time aside to test this. You think you'll have it up today or tomorrow?

**Comment by mananjadhav** [comment]
I had slow day today as well and I'll be able to finish this on weekend. I am just working on a revert. Apologies for the delay.

**Comment by mananjadhav** [comment]
@lakchote 

1. 
> Since the time the DD was written, some changes were made.
> 
> `REJECT_TRANSACTION: 'REJECTTRANSACTION',` -> use now `DECLINEDTRANSACTION_THREAD`
> 
> `REJECTED_TRANSACTION_REMOVEDFROMREPORT: 'REJECTEDTRANSACTIONREMOVEDFROMREPORT'` -> use now `DELETED_TRANSACTION`

There are 4 new constants [here](https://github.com/Expensify/App/pull/62787/files#diff-fccfa43127fe02d958452f91303d6a59a20e60c031c0d7d920ac6175a37c83c0R1424-R1427). Are we supposed to update all?

Also would you mind sharing the reason for the change? especially where one is called declined and the other is called deleted?


2.

>  What did you not understand specifically? Which violation logic in the design doc are you referring to?

The design document has a line:

> move the expense to another report based on Scheduled Submit

<img width="837" alt="Image" src="https://github.com/user-attachments/assets/d8bebbc9-2d2e-40f8-bdff-7eacb7e430f8" />

Can you confirm if this is all the [logic](https://github.com/Expensify/App/pull/62787/files#diff-9c07b44e092fd3d975806649be46d76b75a377c16e32ed86597b630a84c290d0R11221-R11239) that is needed. Or is there any other handling for moving the expense report to be done.


3.
 I did check the method `hasPendingRTERViolation` but I thought we're adding a new violation ` REJECTED_EXPENSE: 'rterRejectedExpense',`.

<img width="887" alt="Image" src="https://github.com/user-attachments/assets/b701c3ff-aedc-4df9-9b1f-bbe3ff304bb8" />

> Maybe follow the same pattern as we are doing in src/libs/actions/IOU.ts
> 
> e.g successData: clear the pending fields, errors, and loading states? And in failureData add the error fields?

My confusion primarily being should we use DELETE pending state or UPDATE pending state?


4. 
> 
>  The correct pattern for updating lastReadTime is:
>    `lastReadTime: DateUtils.getDBTime()`
> 
> This is used throughout the codebase (e.g., src/libs/actions/IOU.ts:1404) when updating report timestamps.

Okay I was using the declineAction created time. Isn't that that correct? Because we'd want the lastReadTime to be before the decline action creation time?

5. 👍 

6.

> There's no specific utility for this action type, but you should follow the pattern of similar functions like [buildOptimisticApprovedReportAction()](https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.ts#L6081) in ReportUtils.ts:6081.

So you're suggesting we create a new method to build this action optimistically?


> 7. That's a good idea to leverage these new components to not repeat logic/code _if_ it can serve our use case perfectly

👍 

> 8. I'd say yes as it handles individual transaction threads.
> 
> This component is used for the header of individual expense/transaction threads, not the overall expense report.

and does that mean `Decline` will also be in the `MoneyRequestHeader` only right?

Yeah the design document mentions `MoneyReportHeader` across. Hence my confusion. So what would the primary action be on `MoneyReportHeader` (overall expense report) when one of the transaction is in declined state?

> 
> 9. You can find them by looking at the git history.
> 

Based on the new flow I think we won't need the flags. But I was using `isReportSubmitter` which is incorrect. Based on your reference I'll check for existing utils and accordingly fix it.

10. The hourglass message will also then move to the transaction thread right? or will it show up in the MoneyReportHeader as well?

<img width="604" alt="Image" src="https://github.com/user-attachments/assets/b38205ac-eac0-46de-8243-db28b2fd5cb4" />

11. In this scenario, we expect the Submit action to be added to the secondary menu. That means we'll need to add additional checks in `ReportPrimaryActionUtils` and `ReportSecondaryUtils` to see if the transaction has declined. But this is on the expense level also right? So do we need any explicit handling for the same?

<img width="857" alt="Image" src="https://github.com/user-attachments/assets/d4e8adc7-79f8-45ad-94e6-139cb89eeea7" />

12. For the system messages, we don't generally add the actor display name right? So here I am assuming we'll only post `marked the decline message as resolved`. And same for others too?

<img width="452" alt="Image" src="https://github.com/user-attachments/assets/3b491e20-1204-46f0-86be-6fef2694b6cb" />

**Comment by mananjadhav** [comment]
> Judging by the code you've linked in your PR, you didn't do this, from what it looks like?

> You always need to create a movedToReportID as the API expects and handles it.



@lakchote I am following the instructions described here and attaching the relevant code with the [same logic](https://github.com/Expensify/App/pull/62787/files#diff-9c07b44e092fd3d975806649be46d76b75a377c16e32ed86597b630a84c290d0R11130-R11247). In this section the moved to report is only for non-instant + multiple expenses. Hence I am building an optimistic report in draft state and moving the expense there. In case of single expense it mentions `or reports with single expense -> Changes report state to DRAFT`. In this scenario what would the movedToReportID be? Because we're updating the current report state to DRAFT?

<img width="852" alt="Image" src="https://github.com/user-attachments/assets/b49fbae4-6337-4548-961a-26aba9e13783" />



> Here are the correct parameters, you're missing some of them:

The parameters are for the Decline Transaction modal. The rest of API call [parameters](https://github.com/Expensify/App/pull/62787/files#diff-9c07b44e092fd3d975806649be46d76b75a377c16e32ed86597b630a84c290d0R11130-R11135) are optimistic. Here's the API [call](https://github.com/Expensify/App/pull/62787/files#diff-9c07b44e092fd3d975806649be46d76b75a377c16e32ed86597b630a84c290d0R11384-R11393). Hence I asked what other checks are needed when I am getting the API errors such as "Report is not in submitted state". I can see you've responded will update.

> > 11. In this scenario, we expect the Submit action to be added to the secondary menu. That means we'll need to add additional checks in `ReportPrimaryActionUtils` and `ReportSecondaryUtils` to see if the transaction has declined. But this is on the expense level also right? So do we need any explicit handling for the same?
> 
> Can you rephrase this please, I'm not sure of understanding what you mean exactly:

Our new UI supports having a Primary Action and More popover menu with secondary actions.

- We decided that "Mark as Resolved" will be at the Transaction thread level `MoneyRequestHeader`.  The expense report can have 0 or more transactions in declined state, is that assumption correct? 
- What would the expense level report primary action be?
- And will the hourglass message needed to be shown at the expense level as well?
- And one of the expectation is to move the `Submitted` to the secondary state, is that for the Transaction thread?
- Lastly, just clarifying, one can decline the transaction at the expense level and mark as resolved at the transaction thread level.


> > I did check the method hasPendingRTERViolation but I thought we're adding a new violation  REJECTED_EXPENSE: 'rterRejectedExpense',.
>
> Yes that's true we're adding a new rterRjectedExpense violation metadata in the transaction's comment NVP.

So are you suggesting we still use `hasPendingRTERViolation` and add the new `rterRjectedExpense` check as well? This will update the behavior at other places where  `hasPendingRTERViolation` being called or continue to keep the scope within the action as done right now.

**Comment by mananjadhav** [comment]
Quick bump for @garrettmknight and @youssef-lr for responses to the comment tagged [here](https://github.com/Expensify/App/issues/52049#issuecomment-2919381972).

**Comment by lakchote** [comment]
@mananjadhav 

The `movedReportID` only applies for this case (for Scheduled Submit with `Instant`):

![Image](https://github.com/user-attachments/assets/2c182dff-1457-47a9-a8be-d2369d735126)

You wouldn't need it for a single expense, as you wouldn't create a new report; you would just change the report's state.

> Our new UI supports having a Primary Action and More popover menu with secondary actions.
> 
> We decided that "Mark as Resolved" will be at the Transaction thread level MoneyRequestHeader. The expense report can have 0 or more transactions in declined state, is that assumption correct?
> What would the expense level report primary action be?
> And will the hourglass message needed to be shown at the expense level as well?
> And one of the expectation is to move the Submitted to the secondary state, is that for the Transaction thread?
> Lastly, just clarifying, one can decline the transaction at the expense level and mark as resolved at the transaction thread level.


I'd like @garrettmknight to answer these questions to confirm these product decisions.

> So are you suggesting we still use hasPendingRTERViolation and add the new rterRjectedExpense check as well? This will update the behavior at other places where hasPendingRTERViolation being called or continue to keep the scope within the action as done right now.

I was just saying that you can use the `hasPendingRTERViolation()` method to know if a transaction has a RTER violation. Optimistically, you should add the `rterRjectedExpense` in the transaction's comment metadata. No need to modify `hasPendingRTERViolation()`.

**Comment by mananjadhav** [comment]
Thanks for the confirmation @lakchote.

**Comment by garrettmknight** [comment]
Our new UI supports having a Primary Action and More popover menu with secondary actions.

- We decided that "Mark as Resolved" will be at the Transaction thread level MoneyRequestHeader. The expense report can have 0 or more transactions in declined state, is that assumption correct?

That's correct.

- What would the expense level report primary action be?
The primary action of a declined expense should be the `Mark as resolved' action.

![Image](https://github.com/user-attachments/assets/294e3a42-df7d-4114-9c6b-643bca984cae)

- And will the hourglass message needed to be shown at the expense level as well?

Correct, the hourglass messages is expense level.

- And one of the expectation is to move the Submitted to the secondary state, is that for the Transaction thread?

The submit action is only available at the expense level for a one-expense report. In the case of a declined transaction on a one expense report, `Submit` should be a secondary action.

If there are more than one expenses on the report, there is no 'Submit' action at the transaction leve.

- Lastly, just clarifying, one can decline the transaction at the expense level and mark as resolved at the transaction thread level.

Both `Decline` and `Mark as resolved` are transaction level actions.

**Comment by garrettmknight** [comment]
@mananjadhav [We're updating the copy on the Hold/Decline modal](https://docs.google.com/document/d/1ukHoMDoX07lwstLn7fBKjWnnopLkDbezQv6Yd6hLIuo/edit?tab=t.0#heading=h.71gukbqqon31). Can you update in your PR?

**Comment by mananjadhav** [comment]
I think I am a bit conflicted when we say

> Both Decline and Mark as resolved are transaction level actions.

> We decided that "Mark as Resolved" will be at the Transaction thread level MoneyRequestHeader

> The primary action of a declined expense should be the `Mark as resolved' action.

Because we've got two components. I thought we wanted to add `Decline` to `MoneyReportHeader` and `Mark as Resoled` to `MoneyRequestHeader`.

Let me go through the screenshots in the previous comments and design doc. I'll make the edit to the code and also send in a video to confirm the behavior.

**Comment by mananjadhav** [comment]
I had all my queries answered with @garrettmknight on DM. Last few questions are also answered on the group thread with @youssef-lr and @lakchote. Thanks, I am unblocked and making progress.

**Comment by mananjadhav** [comment]
@Expensify/design Is this an existing illustration? I can't seem to find it. Also just confirming that the text is bold for the bullet points?

<img width="1075" alt="Image" src="https://github.com/user-attachments/assets/042a7040-0cc6-4e0b-83aa-ff2625b810e8" />

**Implementation:**

<img width="466" alt="Image" src="https://github.com/user-attachments/assets/dfe78560-e072-4ef2-bd49-c0302c039d09" />

**Comment by dubielzyk-expensify** [comment]
> @Expensify/design Is this an existing illustration? I can't seem to find it.

You can find it here: https://github.com/Expensify/App/blob/main/assets/images/simple-illustrations/simple-illustration__realtimereports.svg

>  Also just confirming that the text is bold for the bullet points?

Yeah, I believe that's what we've decided on these kind of modals 👍 


Sidenote: I don't remember seeing this new modal with "hold or decline" (I've only seen the "hold" one). The contrast on the yellow and white `Hold` is very low and I know we spoke about not wanting to have multiple colors for errors, so I wonder if we should actually just have this be a normal headline without any of the colored boxes around it? cc @Expensify/design

**Comment by shawnborton** [comment]
> so I wonder if we should actually just have this be a normal headline without any of the colored boxes around it?

Personally I would be totally fine with that, and I think it would look cleaner anyways. 

> Also just confirming that the text is bold for the bullet points?

I think this is right but I feel like I've maybe seen some of these modals where only some text in each bullet point is bold, to emphasize the particular word(s)? cc @jamesdeanexpensify in case that rings a bell.

**Comment by mananjadhav** [comment]
> I don't remember seeing this new modal with "hold or decline" (I've only seen the "hold" one).

@dubielzyk-expensify This is the new modal design mentioned in the [design doc](https://docs.google.com/document/d/1ukHoMDoX07lwstLn7fBKjWnnopLkDbezQv6Yd6hLIuo/edit?tab=t.0#bookmark=id.wlxa2ibn0vrq).


> I've maybe seen some of these modals where only some text in each bullet point is bold, to emphasize the particular word(s)

I think that is what we had earlier. This seems to be the revised copy and the design. Please check the previous design (ignore the icons, it's copied from the old video)

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/6b4ab329-2af2-4f57-867a-a7b195bc1097" />

**Comment by dubielzyk-expensify** [comment]
> I've maybe seen some of these modals where only some text in each bullet point is bold, to emphasize the particular word(s)

I believe this is what we have had for some of the more onboarding ones too. Don't feel strongly either way.

> @dubielzyk-expensify This is the new modal design mentioned in the [design doc](https://docs.google.com/document/d/1ukHoMDoX07lwstLn7fBKjWnnopLkDbezQv6Yd6hLIuo/edit?tab=t.0#bookmark=id.wlxa2ibn0vrq).

Ahh, thanks. Keen to hear what @dannymcclain thinks

**Comment by dannymcclain** [comment]
Which is the most recent version of the modal? The one with three bullet points or the one with two?

For the three bullet point version, I think it totally makes sense to just use bold for the bullet text. That follows our "empty state card" pattern. The one with two bullet points just has a lot of copy so we needed to do something to establish some visual hierarchy.

>so I wonder if we should actually just have this be a normal headline without any of the colored boxes around it?
>>Personally I would be totally fine with that, and I think it would look cleaner anyways.

Super down with ditching the boxes/colors and just using a regular headline here!

**Comment by dubielzyk-expensify** [comment]
> Super down with ditching the boxes/colors and just using a regular headline here!

I recon we just go for it then. It helps us avoid the color accessibility issues and translations would likely be easier

**Comment by mananjadhav** [comment]
The one with the three points is the latest version.  Team here's what we're discussing then, let me know if this is okay.

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/042cfea0-f909-43c2-a563-6fa208540bd6" />

--------

**Quick update:** I finished the design and the menu changes to add to both `MoneyReportHeader` and `MoneyRequestHeader` yesterday. I'll be working on fixing the API code today.

**Comment by shawnborton** [comment]
That looks good to me assuming it has final copy approval from our marketing folks.

**Comment by mananjadhav** [comment]
Okay I think we should change the subtitle.

It says `If you're not ready to approve or pay an expense, you have two options:` whereas we show 3 bullet points.

**Comment by shawnborton** [comment]
Great catch, cc @garrettmknight for a copy check on that screenshot.

**Comment by dubielzyk-expensify** [comment]
Looks good to me as well (copy aside) 👍

**Comment by jamesdeanexpensify** [comment]
Yeah @garrettmknight I'm not sure if [this](https://github.com/Expensify/App/issues/52049#issuecomment-2957857408) is the latest screenshot. It feels....off. At least copy-wise.

**Comment by jamesdeanexpensify** [comment]
Oh, I see, now I remember we're actually going to show two different screens depending on who's viewing ([GH](https://github.com/Expensify/App/issues/63363)). Great call on updating that lead-in copy @mananjadhav, that was an oversight. 

Maybe we just do: `If you're not ready to approve or pay an expense, you have two options: hold or decline.` and then the bullets come after.

**Comment by garrettmknight** [comment]
I think we can probably just remove the ref to 2 options:
`If you're not ready to approve or pay an expense, you can hold or decline.`

![Image](https://github.com/user-attachments/assets/af3967d3-3289-4620-8719-74bb12b4d1c7)

**Comment by garrettmknight** [comment]
Updated the DD and figma to that in the meantime.

**Comment by jamesdeanexpensify** [comment]
Oh yeah great shout @garrettmknight! The more concise the better.

**Comment by mananjadhav** [comment]
**Update:** Worked on fixing the API calls and the logic based on the recent discussion. Working on setting up the optimistic report actions for different scenarios. The design & copy updates done.

<img width="768" alt="Image" src="https://github.com/user-attachments/assets/a8fbf981-0b63-4045-a225-a7eb87b0ecd3" />

> add it to an existing draft report OR create a new one to add to

@lakchote I am for now creating a new DRAFT report. I hope that is fine.

**Comment by lakchote** [comment]
> @lakchote I am for now creating a new DRAFT report. I hope that is fine.
>

If there's no available report in `DRAFT` state already in the workspace, yes. Otherwise we should add to an existing `DRAFT` report @mananjadhav.

Do you agree @garrettmknight?

**Comment by garrettmknight** [comment]
> If there's no available report in DRAFT state already in the workspace, yes. Otherwise we should add to an existing DRAFT report.

This logic is correct.

**Comment by mananjadhav** [comment]
<img width="888" alt="Image" src="https://github.com/user-attachments/assets/d949cf24-0f17-4048-9604-ed86c55dbaa7" />

@lakchote Can you confirm if this is still true that's mentioned [here](https://github.com/Expensify/App/issues/52049#issuecomment-2910338840)

The response I receive from the backend is `DECLINEDTRANSACTION` and not DELETED as mentioned.

<img width="1179" alt="Image" src="https://github.com/user-attachments/assets/179da475-744a-4e4b-9275-9374940c18b5" />

I was mapping:
- declinedCommentReportActionID: Actual decline reason.
- declinedActionReportActionID: `removed $X.XX from [Merchant Name]`
- which param does `declined this expense` map too?

and I am creating an optimistic report, it's throwing an error:

<img width="1188" alt="Image" src="https://github.com/user-attachments/assets/ecb4449c-eaa6-43c5-b799-025fd4273654" />


and @garrettmknight I was testing  the decline flow.
- If for single expenses we're deleting the report, the submitter won't have an option to Mark as resolved.

**Comment by lakchote** [comment]
@mananjadhav 

> The response I receive from the backend is `DECLINEDTRANSACTION` and not DELETED as mentioned.
> 

I got things mixed up sorry, it's `DECLINEDTRANSACTION` indeed.

> * which param does `declined this expense` map too?
> 

There's no mapping nor a value used for this system message, it's just straight text.


> and I am creating an optimistic report, it's throwing an error:
> 

The error occurs when moving a declined transaction to a target report where the transaction owner differs from the report submitter and lacks write access to that report.

**Comment by garrettmknight** [comment]
> If for single expenses we're deleting the report, the submitter won't have an option to Mark as resolved.

I'm not sure I follow. `Mark as resolved` is a transaction-level violation. Why would deleting the report move it? Also, is this in the Instant Submit case or not?

> **Non-Instant Submit:** When the report has only one expense, we’ll return the report to the Draft state instead of creating a new Draft report to move to and deleting the original Processing report.

This should change the report back to draft.

> If Scheduled Submit is enabled for ‘Instantly’, we’ll remove the expense from the report, and the submitter will be able to access it in their Self DM or via the Reports page.

Can we still maintain the violation at the transaction level if it's not on a report? Is that the issue you're flagging?

**Comment by mananjadhav** [comment]
> There's no mapping nor a value used for this system message, it's just straight text.

The reason I asked is because while in the `Decline` API I get the message but on the submitter OpenReport, I don't get this message in the API response. Hence I thought probably something to do with the param not mapped.

> The error occurs when moving a declined transaction to a target report where the transaction owner differs from the report submitter and lacks write access to that report.

I realized, fixed it.


>  Why would deleting the report move it? Also, is this in the Instant Submit case or not?
> Can we still maintain the violation at the transaction level if it's not on a report? Is that the issue you're flagging?

@garrettmknight For instant submit. I thought we don't show the transaction level violation for instant submit as it goes to Self DM. How does one resolve then?

**Comment by mananjadhav** [comment]
**Update**: Working on fixing the mark as resolved api call. The transaction violation has some bugs and I’ll finish that. 

Decline logic is working fine. Creating a new report is still giving a permission issue as the report I am creating is picking up personal details of the logged in user and not transaction owners. Will fix it.

**Comment by garrettmknight** [comment]
Hmmmm... We'll be changing Instant Submit logic to enable draft reports pretty soon. When that's done this won't be a consideration. Ideally, we could still attempt to add the violation and maybe just silently fail when moving to SelfDM? That way, when we update the logic for Instant Submit to mirror the rest of the frequencies, we don't have to update the violation logic. @lakchote What do you think?

**Comment by mananjadhav** [comment]
Also @lakchote, I am appending the transaction violations with the name `rterRejectedExpense` in the optimistic data but it looks like it's not set as a violation.

<img width="1503" alt="Image" src="https://github.com/user-attachments/assets/33dc922e-4879-48cb-8203-9e5ab5943946" />

The violation that's being set seems to be: `autoReportedRejectedExpense`.

<img width="1364" alt="Image" src="https://github.com/user-attachments/assets/d5f4247d-54e9-4c91-94ab-9531253b2804" />

<img width="1151" alt="Image" src="https://github.com/user-attachments/assets/4e73daa5-b1bd-49ff-ac53-610fda6a1953" />

**Comment by mananjadhav** [comment]
Also :
1. The API still throws an error when it's a single expense with delayed submission. It says `movedToReportID` is mandatory.

2. Not a big problem but the message received on the API is always `new.expensify.com` (Prod URL).

**Comment by mananjadhav** [comment]
Ignore this [comment](https://github.com/Expensify/App/issues/52049#issuecomment-2977542487) but there's a typo in the rejected comment. I can see in the design doc we mention `rterRejectedExpense` but the screenshot in the doc and the API response has `rterRjectedExpense`.

I think we should add it as a violation in the transaction violation object? Otherwise it could end up requiring additional handling everywhere. 

![Image](https://github.com/user-attachments/assets/4443e1e6-962a-4da6-ad21-057ef4c6d06b)

**Comment by garrettmknight** [comment]
Nice catch! 

> I think we should add it as a violation in the transaction violation object?

I'll let @lakchote confirm, but this sounds right to me.

**Comment by lakchote** [comment]
> Also :
> 
> 1. The API still throws an error when it's a single expense with delayed submission. It says `movedToReportID` is mandatory.
> 2. Not a big problem but the message received on the API is always `new.expensify.com` (Prod URL).

1.  If we're in a delayed submission scenario, `movedToReportID` is indeed mandatory as it is right now.
Could you provide a value for this parameter, like the current report? This should put the report back into a draft state.

2.  I don't think this should be a problem; we can always handle that in a follow-up.

---

> I think we should add it as a violation in the transaction violation object?

I'd like to have confirmation from @Beamanator, who wrote the Data storage section. Should we add it there? Thanks!

**Comment by mananjadhav** [comment]
> Could you provide a value for this parameter, like the current report? This should put the report back into a draft state.

Let me test that and see if it works.

>> There's no mapping nor a value used for this system message, it's just straight text.
>>
> The reason I asked is because while in the Decline API I get the message but on the submitter OpenReport, I don't get this message in the API response. Hence I thought probably something to do with the param not mapped.

In case you missed? ^

**Comment by Beamanator** [comment]
>> I think we should add it as a violation in the transaction violation object?
>
> I'd like to have confirmation from @Beamanator, who wrote the Data storage section. Should we add it there? Thanks!

I would think yes, IF we're trying to treat Decline & Reject similarly - I honestly haven't thought about Decline in a bit though :D Are we trying to keep the 2 flows pretty unique or make them line up in NewDot & Classic?

**Comment by mananjadhav** [comment]
> Decline logic is working fine. Creating a new report is still giving a permission issue as the report I am creating is picking up personal details of the logged in user and not transaction owners. Will fix it.

and also this ^. I get an error: `401 Submitter does not own movedToReportID`. As an approver if there are no existing reports, I create one. This is created with the ownerAccountID of the logged-in user. Hence, I can't add the transaction to the report. Locally I do use the submitter's `personalDetails`, so offline data works, but API call doesn't.

**Comment by lakchote** [comment]
> I would think yes, IF we're trying to treat Decline & Reject similarly - I honestly haven't thought about Decline in a bit though :D Are we trying to keep the 2 flows pretty unique or make them line up in NewDot & Classic?

cc @garrettmknight 

>> There's no mapping nor a value used for this system message, it's just straight text.
>>
> The reason I asked is because while in the Decline API I get the message but on the submitter OpenReport, I don't get this message in the API response. Hence I thought probably something to do with the param not mapped.
> In case you missed? ^

I didn't miss it, but I find it weird that we have 2 different messages.
What does the other message look like?

In the DD, there's no link for the action's message, it's just straight text. I think the DD would be the source of truth cc @garrettmknight 


> > Decline logic is working fine. Creating a new report is still giving a permission issue as the report I am creating is picking up personal details of the logged in user and not transaction owners. Will fix it.
> 
> and also this ^. I get an error: `401 Submitter does not own movedToReportID`. As an approver if there are no existing reports, I create one. This is created with the ownerAccountID of the logged-in user. Hence, I can't add the transaction to the report. Locally I do use the submitter's `personalDetails`, so offline data works, but API call doesn't.

Seems like a bug. Do you have the bandwidth to handle it @youssef-lr or not?

**Comment by mananjadhav** [comment]
> > Could you provide a value for this parameter, like the current report? This should put the report back into a draft state.
> 
> Let me test that and see if it works.

@lakchote I tried using the current reportID. It throws an error: `400 movedToReportID is not in the OPEN state`.

**Comment by mananjadhav** [comment]
Attaching both the videos where you'll see the `declined this expense` is removed from the report or not visible on the submitter side. 

https://github.com/user-attachments/assets/944c8e21-5fc9-44f4-87b7-c0e885c74786

https://github.com/user-attachments/assets/2dbceed6-215c-4d0c-aba1-ede7263ddc96

**Comment by garrettmknight** [comment]
Not sure if this is BE or FE, but:
- 'declined this expense' shouldn't be bolded
- the order of the system messages should be:
1. declined this expense
2. comment
3. moved this expense to XYZ report

**Comment by mananjadhav** [comment]
Bold I’ve already fixed on the FE. The order of the message is coming from the BE I think.  I am not adding an optimistic “moved this expense” message.

**Comment by garrettmknight** [comment]
@lakchote I think we'll want to update the order of the messages described [here](https://github.com/Expensify/App/issues/52049#issuecomment-2997301222).

**Comment by lakchote** [comment]
@mananjadhav @garrettmknight, there is a PR that will fix the order of the system messages and update the validation.

There are also frontend fixes to be made in your PR @mananjadhav, mainly:

  1. Wrong report selection - The frontend was finding reports owned by the
   approver instead of the submitter (fixed by adding ownership check)
  2. Wrong report creation - When creating new reports, it was using the
  wrong owner (fixed by using `report.ownerAccountID` instead of
  `actorAccountID`)

Here is the code I had to make it work (in `IOU.ts`):
<details>

```JS
    } else if (hasMultipleExpenses) {
        // For reports with multiple expenses:
        // 1. Update report total
        // 2. Remove expense from report
        // 3. Add to existing draft report or create new one
        const existingOpenReport = Object.values(allReports ?? {}).find((r) => 
            r?.chatReportID === report.chatReportID && 
            r?.type === CONST.REPORT.TYPE.EXPENSE && 
            isOpenReport(r) &&
            r?.ownerAccountID === report.ownerAccountID  // Must be owned by the same submitter
        );

        if (existingOpenReport) {
            movedToReport = existingOpenReport;
            movedToReportID = existingOpenReport.reportID;
            optimisticData.push({
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT}${movedToReport?.reportID}`,
                value: {
                    ...movedToReport,
                    total: (movedToReport?.total ?? 0) + (transaction?.amount ?? 0),
                },
            });
        } else {
            // Generate a new report ID that will be created by the backend
            // The backend will create it with the correct owner (submitter)
            movedToReportID = generateReportID();
        }
        optimisticData.push(
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
                value: {
                    total: (report?.total ?? 0) - (transaction?.amount ?? 0),
                },
            },
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
                value: {
                    reportID: movedToReportID,
                  },
            },
        );

        urlToNavigateBack = ROUTES.REPORT_WITH_ID.getRoute(childReportID);
    } else {
        optimisticData.push({
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
            value: {
                stateNum: CONST.REPORT.STATE_NUM.OPEN,
                statusNum: CONST.REPORT.STATUS_NUM.OPEN,
            },
        });
        urlToNavigateBack = ROUTES.REPORT_WITH_ID.getRoute(report.chatReportID);
    }
```
</details>

Regarding the system message `declined this expense` disappearing when opening the report on the submitter's side, I think this should be handled by the frontend, as the action is already returned by the backend @mananjadhav:

![Image](https://github.com/user-attachments/assets/6cbc9954-5a39-4b9c-87e0-ffa1854f7bb3)

**Comment by mananjadhav** [comment]
>  r?.ownerAccountID === report.ownerAccountID

Yes I am aware and it works my test case. What didn't work was creating a new report.

>  movedToReportID = generateReportID();

@lakchote I think I had this earlier and it didn't work for me. Let me try this again. I thought I was supposed to create a new report locally as well. How do we handle this data optimistically if it's being created at the backend?

and what about the single report scenario where it is expecting movedToReportID? I tried using the same reportID and it didn't work.

**Comment by lakchote** [comment]
I understand your concerns about `generateReportID()` instead of optimistically creating a new report locally.

However,  the previous approach with `createNewReport(personalDetails, policy?.id)` has issues:
It would create a report owned by `actorAccountID` (the approver), not the submitter. Do you have any ideas here?

> and what about the single report scenario where it is expecting movedToReportID? I tried using the same reportID and it didn't work.
>

I've modified the check in the backend so we'd allow not providing one, _if_ there is only a single expense on a delayed submission policy.

@mananjadhav could you check why we're not rendering the "declined this expense" action for the submitter in the meantime in the frontend? Since the action is already returned.

**Comment by lakchote** [comment]
Backend fixes mentioned above have been deployed.

> [@mananjadhav](https://github.com/mananjadhav) could you check why we're not rendering the "declined this expense" action for the submitter in the meantime in the frontend? Since the action is already returned.

friendly bump @mananjadhav

**Comment by mananjadhav** [comment]
I checked the API call and I can see `DECLINEDTRANSACTION_THREAD` and I also have the actionName handling in the `PureReportActionItem` but it's still not rendered. I am debugging it.

**Comment by mananjadhav** [comment]
I also replaced the logic to use `generateReportID` and this is the behavior. I think the parent report tagging is incorrect.

https://github.com/user-attachments/assets/0b3877dc-8fc0-46cd-a0de-60a06a3e8ffa

Also any updates on the transaction violations? I don't see any discussion with respect to that?

<img width="1512" height="493" alt="Image" src="https://github.com/user-attachments/assets/18d61fa8-f88b-455a-be16-ff97356353c3" />

**Comment by mananjadhav** [comment]
I've also tried modifying the `useTransactionViolations` hook to get `rterRjectedExpense` working but I can see that we've added `autoReportedRejectedExpense` to the transaction violations as well.

The issue here is that the data doesn't contain the `comment` and `rejectedBy` keys like it does for `rterRjectedExpense`. We'll need to fix it to be able to work on the generic hook. I've also linked another issue where we're enabling some of these hooks with additional data.


```
{
            "onyxMethod": "mergecollection",
            "key": "transactionViolations_",
            "value": {
                "transactionViolations_1000816485451331211": [
                    {
                        "type": "warning",
                        "name": "autoReportedRejectedExpense",
                        "data": null,
                        "showInReview": true
                    }
                ]
            }
        }

```

**Comment by lakchote** [comment]
@mananjadhav 

> Passing the current reportID for the single expense report with delayed submission still throws the same error. `400 movedToReportID not in OPEN state`
> 

You do not need to provide the `movedReportID` in that case since you are not moving it to a new report; you're only changing the state of the report. I've talked about it [here](https://github.com/Expensify/App/issues/52049#issuecomment-3009094731):

<img width="3760" height="1606" alt="Image" src="https://github.com/user-attachments/assets/ce906f5d-a249-487a-b797-2c7aec1d456f" />

---

> I also replaced the logic to use generateReportID and this is the behavior. I think the parent report tagging is incorrect.
>

Could you please provide a text summary of the expected result and the actual result? So that @garrettmknight can confirm if there are any inconsistencies to fix.

> Also any updates on the transaction violations? I don't see any discussion with respect to that?
>

Which updates specifically? Is that adding the `comment` and `rejectedBy` properties to the `autoReportedRejectedExpense` violation?

There are more than 255 items in this GitHub issue. I'd appreciate it if you could provide a brief context, as I'm often switching up tasks. Thanks!

**Comment by mananjadhav** [comment]
Waiting on some backend changes. Will resume once they're in.

**Comment by garrettmknight** [comment]
Adding @dylanexpensify and @youssef-lr here to keep watch next week: 
- Summary of where we're at is [here](https://expensify.slack.com/archives/C07HPDRELLD/p1754047270353669).
- DM thread for the C+'s is [here](https://expensify.slack.com/archives/C094PPQTDM4/p1753952264996629).
- Flavia's gonna add it to the weekly update next week.

**Comment by dylanexpensify** [comment]
Posting an update on progress today!

**Comment by dylanexpensify** [comment]
[Latest update!](https://expensify.slack.com/archives/C07HPDRELLD/p1754887854027089?thread_ts=1754047270.353669&cid=C07HPDRELLD)

**Comment by mananjadhav** [comment]
PR is ready for review. Will upload the screencasts today.

**Comment by Julesssss** [comment]
An issue [here](https://github.com/Expensify/App/issues/70609) was raised as a deploy blocker -- notifications on Android from reject actions show `Thread` as the title. 

@lakchote although this is a new feature, I think we should fix this or temporarily disable notifications before it hits staging. Let me know if you think otherwise. It should be as simple as changing the title for this notification type. 

<img width="407" height="121" alt="Image" src="https://github.com/user-attachments/assets/ae55eb29-eb9d-4d0c-929a-455db1d56856" />

**Comment by garrettmknight** [comment]
[Latest update here](https://expensify.slack.com/archives/C07NMDKEFMH/p1758644935673199)

**Comment by garrettmknight** [comment]
[Latest update heere](https://expensify.slack.com/archives/C07NMDKEFMH/p1759830849928689)

**Comment by garrettmknight** [comment]
[Latest update here](https://expensify.slack.com/archives/C07NMDKEFMH/p1763406310317039)

**Comment by lakchote** [comment]
I will send the project wrap up tomorrow.
